simpa/core/device_digital_twins/illumination_geometries/ithera_msot_invision_illumination.py

```diff
@@ -13,64 +13,76 @@ class MSOTInVisionIlluminationGeometry(IlluminationGeometryBase):
             angle = 0.0
-        elif self.geometry_id == 1:
+        elif geometry_id == 1:
             angle = 0.0
             det_sep_half = -det_sep_half
             illumination_angle = -illumination_angle
-        elif self.geometry_id == 2:
+        elif geometry_id == 2:
             angle = 1.25664
-        elif self.geometry_id == 3:
+        elif geometry_id == 3:
             angle = 1.25664
             det_sep_half = -det_sep_half
             illumination_angle = -illumination_angle
-        elif self.geometry_id == 4:
+        elif geometry_id == 4:
             angle = -1.25664
-        elif self.geometry_id == 5:
+        elif geometry_id == 5:
             angle = -1.25664
             det_sep_half = -det_sep_half
             illumination_angle = -illumination_angle
-        elif self.geometry_id == 6:
+        elif geometry_id == 6:
             angle = 2.51327
-        elif self.geometry_id == 7:
+        elif geometry_id == 7:
             angle = 2.51327
             det_sep_half = -det_sep_half
             illumination_angle = -illumination_angle
-        elif self.geometry_id == 8:
+        elif geometry_id == 8:
             angle = -2.51327
-        elif self.geometry_id == 9:
+        elif geometry_id == 9:
             angle = -2.51327
             det_sep_half = -det_sep_half
             illumination_angle = -illumination_angle
 
-        source_position = [probe_position_mm[0]/spacing + 1 + np.sin(angle) * detector_iso_distance,
-                           probe_position_mm[1]/spacing + 1 + det_sep_half,
-                           probe_position_mm[2]/spacing + 1 + np.cos(angle) * detector_iso_distance]
+        self.device_position_mm = [self.invision_position[0] + np.sin(angle) * detector_iso_distance,
+                                   self.invision_position[1] + det_sep_half,
+                                   self.invision_position[2] + np.cos(angle) * detector_iso_distance]
+
+        self.source_direction_vector = np.array([-np.sin(angle),
+                                                 np.sin(illumination_angle),
+                                                 np.cos(angle)])
+
+        self.normalized_source_direction_vector = self.source_direction_vector / np.linalg.norm(
+            self.source_direction_vector)
+
+    def get_mcx_illuminator_definition(self, global_settings):
+        self.logger.debug(self.invision_position)
+        source_type = Tags.ILLUMINATION_TYPE_MSOT_INVISION
+
+        spacing = global_settings[Tags.SPACING_MM]
+
+        source_position = list(np.array(self.device_position_mm) / spacing + 1)
 
-        length = np.sqrt(np.sin(angle) ** 2 + np.sin(illumination_angle) ** 2 + np.cos(angle) ** 2)
-        source_direction = [-np.sin(angle) / length,
-                            np.sin(illumination_angle) / length,
-                            np.cos(angle) / length]
+        source_direction = list(self.normalized_source_direction_vector)
```

Comment:
I assume you ran the InVision test that we have?

Reply:
Should I also test the changes on my external scripts as well?

Reply:
Yes I ran the test and it looks good to me. But I think running your external scripts would be a good idea

Reply:
Alright. Will do that tomorrow and let you know!